packages/fx-core/templates/plugins/resource/keyvault/bicep/provision.template.bicep

```diff
@@ -0,0 +1,14 @@
+// Resources for Azure Key Vault
+module keyVaultProvision '{{PluginOutput.fx-resource-key-vault.Provision.keyVault.ProvisionPath}}' = {
+  name: 'keyVaultProvision'
+  params: {
+    provisionParameters: provisionParameters
+    userAssignedIdentityObjectId: {{PluginOutput.fx-resource-identity.References.identityPrincipalId}}
+  }
+}
+
+output keyVaultOutput object = {
+  teamsFxPluginId: 'fx-resource-key-vault'
+  m365ClientSecretReference: keyVaultProvision.outputs.m365ClientSecretReference
```

Comment:
the 2 values will be persisted to state.xxx.json, is it expected? If not, there're 2 options:
1. delete them from plugin context in key vault plugin post provision stage
2. delete the teamsFxPluginId property, or just output the 2 values as 2 different outputs. Only objects with teamsFxPluginId property will be persisted to state.xxx.json.

Reply:
No need to be persisted. Will delete the teamsFxPluginId property.